### PR TITLE
[minor] Static Code Analysis

### PR DESCRIPTION
**Pull request by kalessil**
Followup on https://twitter.com/cakephp/status/989052189304815617

This PR doesn't contain bug-fixes, only code style, control flow and core functions usage tweaks and intends to groom the codebase a little bit.

### CODE REVIEW

File: src/Console/ConsoleIo.php [hunk 1]
```diff
@@ -551,8 +550,7 @@ public function createFile($path, $contents, $forceOverwrite = false)
 
         if (file_exists($path) && $forceOverwrite === false) {
             $this->warning("File `{$path}` exists");
-            $key = $this->askChoice('Do you want to overwrite?', ['y', 'n', 'a', 'q'], 'n');
-            $key = strtolower($key);
+            $key = strtolower($this->askChoice('Do you want to overwrite?', ['y', 'n', 'a', 'q'], 'n'));
```
Comment (dereuromark) [inline src/Console/ConsoleIo.php]: Doing more than a single operation per line is usually considered bad practice.
Harder to debug and easier to overlook issues.

File: src/Core/Plugin.php [hunk 2]
```diff
@@ -241,8 +241,7 @@ public static function loadAll(array $options = [])
             }
         }
         if (Configure::check('plugins')) {
-            $plugins = array_merge($plugins, array_keys(Configure::read('plugins')));
-            $plugins = array_unique($plugins);
+            $plugins = array_unique(array_merge($plugins, array_keys(Configure::read('plugins'))));
```
Comment (dereuromark) [inline src/Core/Plugin.php]: Same, and especially here.

File: src/Core/functions.php [hunk 3]
```diff
@@ -51,7 +51,7 @@ function h($text, $double = true, $charset = null)
             } else {
                 $text = '(object)' . get_class($text);
             }
-        } elseif (is_bool($text) || is_null($text) || is_int($text)) {
+        } elseif (is_bool($text) || is_int($text) || $text === null) {
```
Comment (dereuromark) [inline src/Core/functions.php]: If you want to do the nano opt here, then the null check should be first, not last :)

File: src/Datasource/EntityTrait.php [hunk 4]
```diff
@@ -1356,7 +1356,7 @@ public function source($alias = null)
             get_called_class() . '::source() is deprecated. ' .
             'Use setSource()/getSource() instead.'
         );
-        if (is_null($alias)) {
+        if ($alias === null) {
```
Comment (dereuromark) [inline src/Datasource/EntityTrait.php]: :+1: People should just add my Spryker sniffer rule to the CakePHP rules repo :)

File: src/Filesystem/Folder.php [hunk 5]
```diff
@@ -345,7 +345,7 @@ public static function isAbsolute($path)
      */
     public static function isRegisteredStreamWrapper($path)
     {
-        return preg_match('/^[^:\/\/]+?(?=:\/\/)/i', $path, $matches) &&
+        return preg_match('/^[^:\/\/]+?(?=:\/\/)/', $path, $matches) &&
```
Comment (dereuromark) [inline src/Filesystem/Folder.php]: Can this be a functional regression? isAbsolute() still uses /i for example.

Comment (kalessil) [reply src/Filesystem/Folder.php]: Nope, the regex has no any characters. `i` is not needed.

File: src/View/StringTemplate.php [hunk 6]
```diff
@@ -180,7 +180,7 @@ protected function _compileTemplates(array $templates = [])
             }
 
             $template = str_replace('%', '%%', $template);
-            preg_match_all('#\{\{([\w\d\._]+)\}\}#', $template, $matches);
+            preg_match_all('#\{\{([\w\._]+)\}\}#', $template, $matches);
```
Comment (dereuromark) [inline src/View/StringTemplate.php]: Do we have a test case for this to not introduce an accidental regression?

File: src/Datasource/EntityTrait.php (hunk 4 above)
Comment (kalessil) [reply src/Datasource/EntityTrait.php]: + PHP CS Fixer has `is_null` rule. Hopefully, one of them can be added.

File: src/Core/Plugin.php (hunk 2 above)
Comment (ADmad) [reply src/Core/Plugin.php]: I too don't like this. Reduces code readability.

File: src/Core/functions.php (hunk 3 above)
Comment (kalessil) [reply src/Core/functions.php]: Ideally in separate PR addressing similar cases around the codebase.

File: src/Event/EventManager.php [hunk 7]
```diff
@@ -366,8 +366,7 @@ public function listeners($eventKey)
         $globalListeners = static::instance()->prioritisedListeners($eventKey);
         $globalListeners = empty($globalListeners) ? [] : $globalListeners;
 
-        $priorities = array_merge(array_keys($globalListeners), array_keys($localListeners));
-        $priorities = array_unique($priorities);
+        $priorities = array_unique(array_merge(array_keys($globalListeners), array_keys($localListeners)));
```
Comment (ADmad) [inline src/Event/EventManager.php]: Same comment as above. Too much going on in a single statement.

File: src/View/StringTemplate.php (hunk 6 above)
Comment (kalessil) [reply src/View/StringTemplate.php]: One of the un-testable things: a greedy regex. `\d` is sub-set of `\w`, what only can hurt performance.

File: src/Core/Plugin.php (hunk 2 above)
Comment (kalessil) [reply src/Core/Plugin.php]: Ok, I probably don't have enough context of the project. Will revert this specific inspection fixes.

File: src/Database/Dialect/SqliteDialectTrait.php [hunk 8]
```diff
@@ -120,7 +120,7 @@ protected function _transformFunctionExpression(FunctionExpression $expression)
                     ->setConjunction(' ,')
                     ->iterateParts(function ($p, $key) {
                         if ($key === 0) {
-                            $value = rtrim(strtolower($p), 's');
+                            $value = strtolower(rtrim($p, 's'));
```
Comment (ADmad) [inline src/Database/Dialect/SqliteDialectTrait.php]: This is not functionally same. Earlier a capital `S` at end of string would have been trimmed, now it won't be.

File: src/Core/functions.php (hunk 3 above)
Comment (kalessil) [reply src/Core/functions.php]: OK, gimme a sec

File: src/Database/Dialect/SqliteDialectTrait.php (hunk 8 above)
Comment (kalessil) [reply src/Database/Dialect/SqliteDialectTrait.php]: Hm, good catch. A false positive, reverting.

File: src/Database/Driver/PDODriverTrait.php [hunk 9]
```diff
@@ -158,7 +158,7 @@ public function rollbackTransaction()
             return false;
         }
 
-        return $this->_connection->rollback();
+        return $this->_connection->rollBack();
```
Comment (ADmad) [inline src/Database/Driver/PDODriverTrait.php]: Existing method casing is already correct.

https://github.com/cakephp/cakephp/blob/62944e3bc04bd62459b3fc43c33e06d245837cb1/src/Database/Connection.php#L544

Comment (dereuromark) [reply src/Database/Driver/PDODriverTrait.php]: I wonder how that tool thinks otherwise.

File: src/Core/PluginCollection.php [hunk 10]
```diff
@@ -131,7 +131,7 @@ public function get($name)
      */
     public function next()
     {
-        $this->position += 1;
+        ++$this->position;
```
Comment (markstory) [inline src/Core/PluginCollection.php]: The old form is easier to read in my opinion.

File: src/Database/Driver/PDODriverTrait.php (hunk 9 above)
Comment (chinpei215) [reply src/Database/Driver/PDODriverTrait.php]: The _connection seems to be an instance of PDI. So PDO::rollBack() would be correct.

File: src/Core/PluginCollection.php (hunk 10 above)
Comment (josegonzalez) [reply src/Core/PluginCollection.php]: We use the prefix version? I didn't even know that was a thing in PHP.

Comment (markstory) [reply src/Core/PluginCollection.php]: We do use the suffix form, but generally try to avoid the prefix form as it uncommon.

Comment (markstory) [reply src/Core/PluginCollection.php]: I found 4 other places we are using prefix incrementing. I'll normalize all of the prefix usage after merging this.

Comment (kalessil) [reply src/Core/PluginCollection.php]: Perfect, thank you, Mark.

**Pull request merged**